contracts/connection-router/src/contract/execute.rs

```diff
@@ -137,9 +144,16 @@ where
         sender: &Addr,
         msgs: Vec<Message>,
     ) -> error_stack::Result<Vec<Message>, Error> {
+        let verify_msg_ids = |msgs: &Vec<Message>, expected_format| {
+            msgs.iter()
+                .try_for_each(|msg| msg_id::verify_msg_id(&msg.cc_id.id, expected_format))
+                .change_context(Error::InvalidMessageId)
+        };
```

Comment:
this doesn't close over any variable, so you could also move it into an actual function